### DISCUSSION TITLE
AuthenticationManager.GetACSAppOnlyContext Access Denied

### DISCUSSION THREAD

**Discussion by patrickblanc** [pnp/pnpframework#432] [Q&A] [answered]
Hi everyone
Am I missing something if I told you this.

1. I have my azure ad application with graph Sites.Selected permissions (granted by admin done).
2. I did use PNP PowerShell command 
```powershell
Grant-PnPAzureADAppSitePermission -AppId "<my-app-id>" -DisplayName "<my-app-name>" -Site "<my-sharepoint-site-url>" -Permissions Write
```
3. I am using this code from c# to access "my-sharepoint-site-url"
```c#
PnP.Framework.AuthenticationManager am = new ();
var cc = am.GetACSAppOnlyContext("<my-sharepoint-site-url>", "<my-app-id>", "<my-app-secret>");

cc.Load(cc.Web, w => w.Url);
cc.ExecuteQuery();
``` 

But I got : 

> Microsoft.SharePoint.Client.ServerUnauthorizedAccessException
>   HResult=0x80131500
>   Message=Access is denied

Thanks for your support.

**Comment by jansenbe** [comment, marked as answer]
ACS methods...are only for apps created using SharePoint Azure ACS...When using Azure AD apps (preferred over the old ACS) then you need to configure a certificate with your Azure AD app and use certificate based auth. Doing auth using a secret won't work.
